File: src/com/magayaga/microscript/Function.java
/**
 * MicroScript — The programming language
 * Copyright (c) 2024-2025 Cyril John Magayaga
 * 
 * It was originally written in Java programming language.
 */
package com.magayaga.microscript;

import java.util.List;

public class Function {
    private final String name;
    private final List<Parameter> parameters;
    private final String returnType;
    private final List<String> body;

    public Function(String name, List<Parameter> parameters, String returnType, List<String> body) {
        this.name = name;
        this.parameters = parameters;
        this.returnType = returnType;
        this.body = body;
    }

    public String getName() {
        return name;
    }

    public List<Parameter> getParameters() {
        return parameters;
    }

    public String getReturnType() {
        return returnType;
    }

    public List<String> getBody() {
        return body;
    }
}

File: src/function.go
/**
 * MicroScript — The programming language
 * Copyright (c) 2024, 2025 Cyril John Magayaga
 *
 * It was originally written in Go programming language
 */
package main

// Function represents a function definition in MicroScript
type Function struct {
	Name       string
	Parameters []Parameter
	ReturnType string
	Body       []string
}

// NewFunction creates a new function
func NewFunction(name string, parameters []Parameter, returnType string, body []string) *Function {
	return &Function{
		Name:       name,
		Parameters: parameters,
		ReturnType: returnType,
		Body:       body,
	}
}

// GetName returns the function name
func (f *Function) GetName() string {
	return f.Name
}

// GetParameters returns the function parameters
func (f *Function) GetParameters() []Parameter {
	return f.Parameters
}

// GetReturnType returns the function return type
func (f *Function) GetReturnType() string {
	return f.ReturnType
}

// GetBody returns the function body
func (f *Function) GetBody() []string {
	return f.Body
}
